### PR TITLE
displays orders by date

### PR DESCRIPTION
**Pull request by g1er**
Displays all orders of chosen date in Content

### CODE REVIEW

File: src/app/cart/components/all-orders/all-orders-content/all-orders-content.component.ts [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
-  templateUrl: './all-orders-content.component.html',
-  styleUrls: ['./all-orders-content.component.scss']
+  selector: "all-orders-content",
+  templateUrl: "./all-orders-content.component.html",
+  styleUrls: ["./all-orders-content.component.scss"],
 })
 export class AllOrdersContentComponent implements OnInit {
-  @Input() allOrdersDetails
+  @Input() allOrdersDetails;
+  @Input() ordersOnDate = [];
```
Comment (IgorKulishov) [inline src/app/cart/components/all-orders/all-orders-content/all-orders-content.component.ts:10]: can you please rename to "ordersForDate"

File: src/app/cart/components/all-orders/all-orders-content/all-orders-content.component.ts [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
   accordionPosition: { [index: number]: boolean } = [];
-  constructor() { }
-
-  ngOnInit(): void {
-  }
+  constructor() {}
 
+  ngOnInit(): void {}
```
Comment (IgorKulishov) [inline src/app/cart/components/all-orders/all-orders-content/all-orders-content.component.ts]: It's not used - we can remove it (with import)

File: src/app/cart/components/all-orders/all-orders-filter/all-orders-filter.component.html [hunk 3]
```diff
@@ -1 +1,4 @@
-<p>all-orders-filter works!</p>
+<div *ngFor='let date of orderDates'>
+    <input type="radio" id="scales" name="date" value='{{date}}' (change)="onSelectDate(date)">
```
Comment (IgorKulishov) [inline src/app/cart/components/all-orders/all-orders-filter/all-orders-filter.component.html]: can we rename `id="scales"` to `id="date-0"`, `id="date-1"`, etc:

For that we need to:
1) add index inside *ngFor 
`*ngFor="let item of items; index as i;`
Source: https://angular.io/api/common/NgForOf

2)  `id="date-{{i}}"`

File: src/app/cart/components/all-orders/all-orders-filter/all-orders-filter.component.ts [hunk 4]
```diff
@@ -1,15 +1,18 @@
...
 })
 export class AllOrdersFilterComponent implements OnInit {
+  @Input() orderDates = [];
+  @Output() selectDate = new EventEmitter<string>();
 
-  constructor() { }
+  constructor() {}
 
-  ngOnInit(): void {
+  ngOnInit(): void {}
```
Comment (IgorKulishov) [inline src/app/cart/components/all-orders/all-orders-filter/all-orders-filter.component.ts]: we don't need `ngOnInit(): void {}`

File: src/app/cart/components/all-orders/all-orders.component.ts [hunk 5]
```diff
@@ -14,15 +15,25 @@ import { CartState } from "../../store/states/cart.states";
...
+    private store: Store<CartState>
+  ) {
     this.allOrdersDetails$ = this.store
       .select(selectAllOrdersHistory)
       .pipe(map((store) => store));
   }
 
   ngOnInit(): void {
+    this.orderDates$ = this.ordersHistory.fetchOrdersDatesForCurrentMonth(
+      "2021-08"
```
Comment (IgorKulishov) [inline src/app/cart/components/all-orders/all-orders.component.ts]: Please create a new method to return "currentYear-currentMonth" in format YYYY-MM (months start from 01 to 12) and replace the hardcoded value.
If month is 1-9  it should be with "0" at the front: e.g. 01-09.

File: src/app/cart/service/orders-history-api.service.ts [hunk 6]
```diff
@@ -1,99 +1,154 @@
...
-        headers: headers
+        headers: headers,
       };
-      return this.http.get(environment.ORDER_HISTORY_API + `/orders`, options );
+      return this.http.get(environment.ORDER_HISTORY_API + `/orders`, options);
     }
   }
-  saveOrder() {
+  fetchOrdersDatesForCurrentMonth(yearMonth: string): Observable<any> {
+    yearMonth = yearMonth || "2021-08";
```
Comment (IgorKulishov) [inline src/app/cart/service/orders-history-api.service.ts]: Please remove hardcoded value. That was temporary - we don't need it.

File: src/app/cart/components/all-orders/all-orders-content/all-orders-content.component.html [hunk 7]
```diff
@@ -10,6 +10,11 @@
       <tbody *ngFor='let allOrdersDetail of allOrdersDetails' id="order-block-for-desktop">
       <div class="all-block-item">{{allOrdersDetail | json}}</div>
       </tbody>
+
+      <div *ngFor="let orderOnDate of ordersOnDate">
```
Comment (IgorKulishov) [inline src/app/cart/components/all-orders/all-orders-content/all-orders-content.component.html:14]: That is temporary code - you can keep it for now
Later we will need to work on the component

File: src/app/cart/components/all-orders/all-orders-content/all-orders-content.component.ts (hunk 2 above)
Comment (g1er) [reply src/app/cart/components/all-orders/all-orders-content/all-orders-content.component.ts]: done

File: src/app/cart/components/all-orders/all-orders-filter/all-orders-filter.component.html (hunk 3 above)
Comment (g1er) [reply src/app/cart/components/all-orders/all-orders-filter/all-orders-filter.component.html]: done

File: src/app/cart/components/all-orders/all-orders-filter/all-orders-filter.component.ts (hunk 4 above)
Comment (g1er) [reply src/app/cart/components/all-orders/all-orders-filter/all-orders-filter.component.ts]: done

File: src/app/cart/service/orders-history-api.service.ts (hunk 6 above)
Comment (g1er) [reply src/app/cart/service/orders-history-api.service.ts]: Done

File: src/app/cart/components/all-orders/all-orders.component.ts [hunk 8]
```diff
@@ -17,19 +18,26 @@ export class AllOrdersComponent implements OnInit {
...
+    this.currentYearMonth = this.datePipe.transform(
+      this.currentYearMonth,
+      "yyyy-MM"
+    );
   }
 
   ngOnInit(): void {
     this.orderDates$ = this.ordersHistory.fetchOrdersDatesForCurrentMonth(
-      "2021-08"
+      this.currentYearMonth
```
Comment (IgorKulishov) [inline src/app/cart/components/all-orders/all-orders.component.ts:40]: This one is good, I did not know you can do it

Comment (IgorKulishov) [reply src/app/cart/components/all-orders/all-orders.component.ts:40]: we can add unit test for that logic

**Review by IgorKulishov — APPROVED** [review]

**Pull request merged**